src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -8,12 +8,31 @@
 
 namespace System.Buffers
 {
-    public unsafe sealed class NativeBufferPool<T> where T : struct
+    public unsafe sealed class NativeBufferPool<T> : IDisposable where T : struct
     {
         private const int NUMBER_OF_BUFFERS_IN_BUCKET = 50;
         private int _maxElements;
         private NativeBufferBucket<T>[] _buckets;
 
+        private static NativeBufferPool<byte> _sharedInstance = null;
+
+        public static NativeBufferPool<byte> SharedByteBufferPool
+        {
```

Comment:
how does this work?

alloc new pool if shared reference is null, assign pool to shared reference, **dispose pool**, return reference?

What am I missing?

Are we attempting to guard against a race condition causing multiple pools to be allocated? If so, then add an actual guard. Either use a mutex or set a value that an allocation is happening. I prefer a mutex because the second caller will block and always successfully return.